src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,78 @@
+package duke;
+
+/**
+ * Represents a class which contains methods to make sense of the commands.
+ */
+public class Parser {
+    String stringToParse;
+
+    /**
+     * Creates a Parser object.
+     * @param stringToParse String command to be parse to capture the keyword.
+     */
+    Parser(String stringToParse) {
+        this.stringToParse = stringToParse;
+    }
+
+    /**
+     * Creates a new command depending if it has the correct substring for that command.
+     * @return Command depending on the substrings of different commands.
+     * @throws DukeUnknownWordException
+     */
+    Command parse() throws DukeUnknownWordException {
+        if (isSubstringForDeleteCommand()){
+            return new DeleteCommand();
+        } else if (isSubstringForMarkCommand()){
+            return new MarkCommand();
+        } else if (isSubstringForUnmarkCommand()){
+            return new UnmarkCommand();
+        } else if (isSubstringForDeadlineCommand()){
+            return new DeadlineCommand();
+        } else if (isSubstringForEventCommand()) {
+            return new EventCommand();
+        } else if (isSubstringForTodoCommand()){
+            return new TodoCommand();
+        } else if (isSubstringForShowListCommand()){
+            return new ShowListCommand();
+        } else if (isSubstringForFindCommand()) {
+            return new FindCommand();
+       // } else if (isSubStringForExitCommand()){
+         //   return new ExitCommand();
+        } else {
+            throw new DukeUnknownWordException();
+        }
+    }
```

Comment:
I like how you put the checking of a command in other methods to make ```parse``` easier to read 👍